src/Tools/dotnet-monitor/EndpointInfo/ServerEndpointInfoSource.cs

```diff
@@ -77,11 +82,45 @@ internal sealed class ServerEndpointInfoSource : BackgroundService, IEndpointInf
             _pendingRemovalWriter = pendingRemovalChannel.Writer;
         }
 
-        public override void Dispose()
+        public async ValueTask DisposeAsync()
         {
-            base.Dispose();
+            if (!DisposableHelper.CanDispose(ref _disposalState))
+                return;
+
+            // Makes sure the background task is canceled.
+            Dispose();
 
             _pendingRemovalWriter.TryComplete();
+
+            // Wait for background services to complete.
+            await ExecuteTask.SafeAwait().WaitAsync(CancellationToken.None);
```

Comment:
Just double checking that we want `CancellationToken.None` for these - I know we typically avoid doing that.

Reply:
We don't have access to a specified token and creating our own `CancellationTokenSource` will likely not fire in time when shutting down the tool. I think the `ExecuteTask` is going to complete fairly quickly and "always" complete. But what you've pointed out is a good thing to remember in general.